src/screens/Main/EditStack/SnippetSelection/SnippetSelection.tsx

```diff
@@ -0,0 +1,175 @@
+                    if(selectedQuestions[item.id-1].isSelected){
+                      const temp = []
+                      const preOrder = item.orderInList
+                      for(let i = 0; i < selectedQuestions.length; i++){
+                        if(i != (item.id-1)){
+                          if(selectedQuestions[i].orderInList > preOrder){  
+                            const newQuestion: QuestionSelected = {
+                              id: selectedQuestions[i].id, 
+                              isSelected: selectedQuestions[i].isSelected, 
+                              orderInList: selectedQuestions[i].orderInList-1,
+                              text: selectedQuestions[i].text
+                            }
+                            temp.push(newQuestion)
+                          } else{
+                            const newQuestion: QuestionSelected = {
+                              id: selectedQuestions[i].id, 
+                              isSelected: selectedQuestions[i].isSelected, 
+                              orderInList: selectedQuestions[i].orderInList,
+                              text: selectedQuestions[i].text
+                            }
+                            temp.push(newQuestion)
+                          }
+                        } else{
+                          const newQuestion: QuestionSelected = {
+                            id: item.id, 
+                            isSelected: false, 
+                            orderInList: 0,
+                            text: item.text
+                          }
+                          temp.push(newQuestion)
+                        }
+                        setSelectedQuestions(temp)
+                      }
+                    } else{
+                      const temp = []
+                      for(let i = 0; i < selectedQuestions.length; i++){
+                        if(i != (item.id-1)){
+                          const newQuestion: QuestionSelected = {
+                            id: selectedQuestions[i].id, 
+                            isSelected: selectedQuestions[i].isSelected, 
+                            orderInList: selectedQuestions[i].orderInList,
+                            text: selectedQuestions[i].text
+                          }
+                          temp.push(newQuestion)
+                        } else{
+                          const newQuestion: QuestionSelected = {
+                            id: item.id, 
+                            isSelected: true, 
+                            orderInList: numSelected,
+                            text: item.text
+                          }
+                          temp.push(newQuestion)
+                        }
+                        setSelectedQuestions(temp)
+                      }
+                    }
+
+                    if(item.isSelected){
+                      setNumSelected(numSelected-1)
```

Comment:
can you do this above when you handle the selection 
then all you check is numSelected